pkg/gardenlet/operation/botanist/operatingsystemconfig.go

```diff
@@ -110,6 +110,13 @@ func (b *Botanist) DeployOperatingSystemConfig(ctx context.Context) error {
 	b.Shoot.Components.Extensions.OperatingSystemConfig.SetAPIServerURL(fmt.Sprintf("https://%s", b.Shoot.ComputeOutOfClusterAPIServerAddress(true)))
 	b.Shoot.Components.Extensions.OperatingSystemConfig.SetCABundle(b.getOperatingSystemConfigCABundle(clusterCASecret.Data[secretsutils.DataKeyCertificateBundle]))
 
+	shoot := b.Shoot.GetInfo()
+	var credRotStatus *gardencorev1beta1.ShootCredentialsRotation
+	if shoot != nil && shoot.Status.Credentials != nil {
+		credRotStatus = shoot.Status.Credentials.Rotation
+	}
+	b.Shoot.Components.Extensions.OperatingSystemConfig.SetCredentialsRotationStatus(credRotStatus)
+
```

Comment:
Why not make it more concise? The `nil` check for shoot is not necessary. It is already assumed to be non-nil prior to this code, e.g. in `DefaultOperatingSystemConfig` or directly after this in `ShootEnablesSSHAccess(...)`. Calling `SetCredentialRotationStatus` with `nil` as parameter does not seem to be required either.

```suggestion
	shoot := b.Shoot.GetInfo()
	if shoot.Status.Credentials != nil {
		b.Shoot.Components.Extensions.OperatingSystemConfig.SetCredentialsRotationStatus(shoot.Status.Credentials.Rotation)
	}
	

```

Reply:
Oh, indeed. Thanks!